packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -302,8 +302,13 @@ export const getSectionVisibleValue = (
   // If you press `1`, on the first section, the new value is also `1/dd/yyyy`,
   // So the browser will not fire the input `onChange`.
   // Adding the ltr mark is not a problem because it's only for digit (which are always ltr)
-  // The \u2068 and \u2069 are cleaned, but not the \u200e to notice that an update with same digit occures
-  if (willBeRenderedInInput && section.contentType === 'digit' && !section.hasTrailingZeroes) {
+  // The \u2068 and \u2069 are cleaned, but not the \u200e to notice that an update with same digit occurs
+  if (
+    willBeRenderedInInput &&
+    section.contentType === 'digit' &&
+    !section.hasTrailingZeroes &&
+    value.length === 1
```

Comment:
If the previous value is more than 1 character long, then the pressed character can not be equal to the previous value.
So the invisible character is not useful.

Not putting it when not needed simplifies the tests because we only need to set the invisible character on our `onChange` value in a few occasions rather than everytime.